cin_validator/rules/cin2022_23/rule_4180.py

```diff
@@ -35,11 +35,13 @@ def validate(
 
     # implement rule logic as described by the Github issue. Put the description as a comment above the implementation as shown.
 
-    valid_gender_codes = [1, 2, 0, 9]
+    valid_gender_codes = [1, 2, 0, 9, "1", "2", "0", "9"]
```

Comment:
Since you are doing df.astype(str), you can expect all values at the comparison stage to be in string format.